wear/src/main/kotlin/io/homeassistant/companion/android/data/WearDns.kt

```diff
@@ -0,0 +1,108 @@
+package io.homeassistant.companion.android.data
+
+import android.content.Context
+import com.google.android.gms.wearable.CapabilityClient
+import com.google.android.gms.wearable.Wearable
+import dagger.hilt.android.qualifiers.ApplicationContext
+import java.net.InetAddress
+import java.net.UnknownHostException
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+import java.util.concurrent.ConcurrentHashMap
+import javax.inject.Inject
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.tasks.await
+import okhttp3.Dns
+
+/**
+ * Wear specific implementation of Dns that defaults to the System DNS,
+ * but can fall back to relying on mobile Dns to resolve dns issues.
+ */
+class WearDns @Inject constructor(
+    @ApplicationContext private val appContext: Context
+) : Dns {
+    private val dnsHelperCache = ConcurrentHashMap<String, CacheResult>()
+
+    override fun lookup(hostname: String): List<InetAddress> {
+        return try {
+            Dns.SYSTEM.lookup(hostname)
+        } catch (e: Exception) {
+            if (couldBeWearDnsIssue(e)) {
+                val result = runBlocking { attemptLookupViaMobile(hostname, e) }
+
+                if (result != null) {
+                    return result
+                }
+            }
+
+            throw e
+        }
+    }
+
+    private suspend fun attemptLookupViaMobile(
+        hostname: String,
+        e: Exception
+    ): List<InetAddress>? {
```

Comment:
```suggestion
    private suspend fun attemptLookupViaMobile(
        hostname: String
    ): List<InetAddress> {
```

I think this function should not know anything about the previous failure and always return a list or throw.